scipy/spatial/distance.py

```diff
@@ -1256,16 +1259,20 @@ def jensenshannon(p, q, base=None):
     0.46450140402245893
     >>> distance.jensenshannon([1.0, 0.0, 0.0], [1.0, 0.0, 0.0])
     0.0
+    >>> distance.jensenshannon([[1.0, 2.0]], [[0.5, 1.5]], axis=0)
+    array([0.0, 0.0])
+    >>> distance.jensenshannon([[1.0, 2.0]], [[0.5, 1.5]], axis=1)
+    array([0.0649045])
 
     """
     p = np.asarray(p)
     q = np.asarray(q)
```

Comment:
I would add a validation of the shapes here. Now it's more complex with the axis and could help the user.

Reply:
Not sure that's a good idea. NumPy will give an appropriate error on shape mismatch and reimplementing the correct checks here, keeping in mind broadcasting would be fairly complicated/another source of bugs.

Reply:
Ok closing this then.